fileio/ft_read_event.m

```diff
@@ -2022,6 +2022,47 @@
       hdr = ft_read_header(filename);
     end
     event = read_spmeeg_event(filename, 'header', hdr);
+  
+  case {'blackrock_nev', 'blackrock_nsx'}
+    % use the NPMK toolbox for the file reading
+    ft_hastoolbox('NPMK', 1);
+    
+    % ensure that the filename contains a full path specification,
+    % otherwise the low-level function fails
+    [p,f,e] = fileparts(filename);
+    if ~isempty(p)
+      % this is OK
+    elseif isempty(p)
+      filename = which(filename);
+		end
+		
+    % 'noread' prevents reading of the spike waveforms 
+    % 'nosave' prevents the automatic conversion of
+    % the .nev file as a .mat file
+    orig = openNEV(filename, 'noread', 'nosave')
+
```

Comment:
This code should work. I would delete lines 2044 to 2047 and 2054 to 2057. 
In line 2062, it should say:
event(k).value     = eventCodes(k);

This will be the event codes coming from the serial-digital-IO. There are no codes associated with "Comments", "Spikes", etc. These are coded in separate fields of the orig.Data structure. The field orig.Data.SerialDigitalIO.InsertionReason does NOT dissociate between those codes. Unfortunately, I cannot contribute any clever way to implement more here because I don't have data that contains "comments" or "spikes". 

One crucial point: The .nev files (read in here) will always contain the event codes with 30K sampling but the actual data (.nsx files) can come with different sampling rates. I take care of this in a custom-written 'blackrock_to_ft_trialfun' which can be called like this 

filename1 = 'session1.nev'; % event file
filename2 = 'session2.ns2'; % different file formats (.ns2, .ns6, etc.) have different sampling freqs

cfg = [];
cfg.dataset = filename2;
cfg.eventfile = filename;
cfg.trialdef.eventtype='trigger';
cfg.trialdef.eventvalue=1;
cfg.trialdef.prestim        = 1;
cfg.trialdef.poststim       = 1;
cfg.trialfun                = 'blackrock_to_ft_trialfun';
cfg                         = ft_definetrial(cfg); 

%% 
The trial function looks like this:

function trl=blackrock_to_ft_trialfun(cfg)
hdr=ft_read_header(cfg.dataset, 'headerformat','blackrock_nsx');
event = ft_read_event(cfg.eventfile, 'eventformat', 'blackrock_nev');
trl=[];

 for i=1:length(event)
  if strcmp(event(i).type, 'trigger')
    % it is a trigger, see whether it has the right value
    if ismember(event(i).value, cfg.trialdef.eventvalue)
      % add this to the trl definition
      begsample     = event(i).sample*hdr.Fs - cfg.trialdef.prestim*hdr.Fs;
      endsample     = event(i).sample*hdr.Fs + cfg.trialdef.poststim*hdr.Fs - 1;
      offset        = -cfg.trialdef.prestim*hdr.Fs;  
      trigger       = event(i).value; % remember the trigger (=condition) for each trial
      trl(end+1, :) = [round([begsample endsample offset])  trigger]; 
    end
  end
end